src/event/player/PlayerCreationEvent.php

```diff
@@ -109,4 +116,20 @@ public function setPlayerClass(string $class) : void{
 		Utils::testValidInstance($class, $this->baseClass);
 		$this->playerClass = $class;
 	}
+
+	/**
+	 * The typical use cases for this WaitGroup include:
+	 *  - Carrying out preparatory tasks such as configuring default preferences, setting
+	 * player-specific settings, or initializing in-game assets.
+	 *  - Loading initial player data from external sources or databases before the login sequence begins.
+	 *  - Providing a tailored welcome experience, which might involve sending messages,
+	 * granting starter items, or awarding initial rewards.
+	 *
+	 * Note:
+	 *  After each task is completed, it's essential to call the `done` method of the WaitGroup
+	 *  to signal its completion and allow the WaitGroup to determine when to proceed.
+	 */
```

Comment:
A brief summary line at the top would be good, something like:

`Returns a WaitGroup which allows pausing the player creation until a given number of tasks have been completed`